x/epochstorage/README.md

```diff
@@ -0,0 +1,172 @@
+
+when a new epoch starts, the module will create a copy of the current stake and save it as a specific epoch stake storage. this is done in the begin block of the module.
+
+also, when a new epoch starts the epoch storage will delete any outdated stakestorage (determined by the param EpochsToSave).
+note that if a stakestorage does not exist (either if it was deleted or it is in the future) pairing, verify pairing and relay payments will fail for the requested epoch (look at pairing readme).
+
+### StakeEntry
+
+The stake entry is a struct that contains all the information of a provider.
+
+```go
+type StakeEntry struct {
+	Stake              types.Coin // the providers stake amount (self delegation)
+	Address            string     // the lava address of the provider
+	StakeAppliedBlock  uint64     // the block at which the provider is included in the pairing list
+	Endpoints          []Endpoint // the endpoints of the provider
+	Geolocation        int32      // the geolocation this provider supports
+	Chain              string     // the chainID of the provider
+	Moniker            string     // free string description
+	DelegateTotal      types.Coin // total delegation to the provider (without self delegation)
+	DelegateLimit      types.Coin // delegation total limit
+	DelegateCommission uint64     // commision from delegation rewards
+}
+```
+
+Geolocation are bit flags that indicate all the geolocations that the provider supports, this is the sum of all endpoints geolocations.
+for more about [geolocation](../../proto/lavanet/lava/plans/plan.proto)
+
+for more information about delegation, go to dualstaking README.md
+
+### EndPoint
+
+Endpoint is a struct that describes how to connect and receive services from the provider for the specific interface and geolocation
+
+```go
+type Endpoint struct {
+	IPPORT        string   // the ip and port where the consumer can access the provider
+	Geolocation   int32    // the geolocation of the endpoint
+	Addons        []string // the supported addons of the endpoint
+	ApiInterfaces []string // the supported interfaces by the enpoint
+	Extensions    []string // the supported extensions of the endpoint
+}
+```
+
+for more info on `Addons` and `Extensions` look at the spec module README.md
+
+## Parameters
+
+The epochstorage parameters:
+
+| Key                                    | Type                    | Default Value    |
+| -------------------------------------- | ----------------------- | -----------------|
+| EpochBlocks                            | uint64                  | 30               |
+| EpochsToSave                           | uint64                  | 20               |
+| LatestParamChange                      | uint64                  | NA               |
+
+`EpochBlocks` determines the amount of blocks for each epoch
```

Comment:
put extra new line to make the sentences not appear right after another. Also, put "." on the end of sentences

Reply:
done

Reply:
I meant between EpochBlocks line, EpochsToSave line and LatestParamChange line. The problem is that if you look at how the readme is generated, those lines come one after another without new lines at all (to indicate new line in markdown, you need to put double space)

Reply:
got it, done